Objects/typeobject.c

```diff
@@ -267,7 +268,10 @@ _PyType_ClearCache(struct type_cache *cache)
 #endif
 
     unsigned int cur_version_tag = next_version_tag - 1;
-    next_version_tag = 0;
+    if (_Py_IsMainInterpreter(interp)) {
+        next_version_tag = 0;
```

Comment:
It would be safer to have a per-interpreter next_version_tag. The problem is that there are still many static types which are shared between interpreters :-/ You may add a comment explaining that:

```
// bpo-42745: next_version_tag remains shared by all interpreters because of static types
```

Or put the comment on the variable definition:

```
// Used to set PyTypeObject.tp_version_tag
static unsigned int next_version_tag = 0;
```

Reply:
Thanks, ok i added